cmd/stack_lint.go

```diff
@@ -0,0 +1,135 @@
+	Use:   "lint",
+	Short: "Lint your stack to verify that it conforms to the standard of an Appsody stack",
+	Long: `This command will validate that your stack has the structure of an Appsody stack. It will inform you of files/directories
+missing and warn you if your stack could be enhanced.
+
+This command can be run from the base directory of your stack or you can supply a path to the stack as an argument.`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+
+		stackPath := os.Getenv("PWD")
+		errorCount := 0
+		warningCount := 0
+
+		if len(args) > 0 {
+			stackPath = args[0]
+		}
+
+		imagePath := stackPath + "/image"
+		templatePath := stackPath + "/templates"
+		configPath := imagePath + "/config"
+		projectPath := imagePath + "/project"
+
+		Info.log("LINTING " + path.Base(stackPath) + "\n")
+
+		if fileDoesNotExist(stackPath+"/README.md") != nil {
+			Info.log("ERROR: Missing README.md in: " + stackPath)
+			errorCount++
+		}
+
+		if fileDoesNotExist(stackPath+"/stack.yaml") != nil {
+			Info.log("ERROR: Missing stack.yaml in: " + stackPath)
+			errorCount++
+		}
+
+		if fileDoesNotExist(imagePath) != nil {
+			Info.log("ERROR: Missing image directory in " + stackPath)
+			errorCount++
+		}
+
+		if fileDoesNotExist(imagePath+"/Dockerfile-stack") != nil {
+			Info.log("ERROR: Missing Dockerfile-stack in " + imagePath)
+			errorCount++
+		}
+
+		if fileDoesNotExist(imagePath+"/LICENSE") != nil {
+			Info.log("ERROR: Missing LICENSE in " + imagePath)
+			errorCount++
+		}
+
+		if fileDoesNotExist(configPath) != nil {
+			Info.log("WARNING: Missing config directory in " + imagePath + " (Knative deployment will be used over Kubernetes)")
+			warningCount++
+
+		}
+
+		if fileDoesNotExist(configPath+"/app-deploy.yaml") != nil {
+			Info.log("WARNING: Missing app-deploy.yaml in " + configPath + " (Knative deployment will be used over Kubernetes)")
+			warningCount++
+		}
+
+		if fileDoesNotExist(projectPath+"/Dockerfile") != nil {
+			Info.log("WARNING: Missing Dockerfile in " + projectPath)
+			warningCount++
+		}
+
+		if fileDoesNotExist(templatePath) != nil {
+			Info.log("ERROR: Missing template directory in: " + stackPath)
+			errorCount++
+		}
+
+		if IsEmptyDir(templatePath) != nil {
+			Info.log("ERROR: No templates found in: " + templatePath)
+			errorCount++
+		}
+
+		templates, _ := ioutil.ReadDir(templatePath)
+		for _, f := range templates {
+			if fileDoesNotExist(templatePath+"/"+f.Name()+"/"+".appsody-config.yaml") == nil {
+				Info.log("ERROR: Unexpected .appsody-config.yaml in " + templatePath + "/" + f.Name())
+				errorCount++
+			}
+		}
+
+		if errorCount > 0 {
+			Info.log("\nLINT TEST FAILED")
+			Info.log("\nTOTAL ERRORS: " + strconv.Itoa(errorCount))
+			Info.log("TOTAL WARNINGS: " + strconv.Itoa(warningCount))
+
+		} else {
+			Info.log("\nLINT TEST PASSED")
+			Info.log("TOTAL WARNINGS: " + strconv.Itoa(warningCount))
+		}
+
+		return nil
+	},
+}
+
+func IsEmptyDir(name string) error {
```

Comment:
Ideally - these would go in utils.go - but since we are creating a new package, you then would have to import `cmd` (or we could move utils to their own package, which would have to be imported by everyone)